openedx/core/djangoapps/user_authn/docs/decisions/0001_simplify_login_ratelimiting.rst

```diff
@@ -0,0 +1,95 @@
+Simplify Login Rate Limiting
+============================
+
+Status
+------
+
+Accepted
+
+Context
+-------
+
+1st party vs 3rd party login
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+edx-platform allows for both 1st party auth, where you provide the LMS with
+your credentials and it gives you back some session tokens, and also 3rd party
+auth, in which you are directed to a 3rd party to authenticate and then
+redirected to the LMS with a token from that third party which is exchanged for
+1st party(LMS) session tokens.
+
+Login User View
+~~~~~~~~~~~~~~~
+
+..
+    TODO: See if I can make this a link instead.
+The ``login_user`` view in ``views/login.py`` is called for both 1st and 3rd
+party login flows.  In the 3rd party login flow, it's called as the callback
+when redirected from the 3rd party login back to the LMS.
+
+Currently this view is accessed through two different endpoints.  It is
+currently a pass-through call for the ``/api/user/v1/account/login_session``
+using a DRF view.  It is also called directly at ``/login_ajax``
+
+Rate limiting Logins
+~~~~~~~~~~~~~~~~~~~~
+
+Currently there are five different rate limiting implementations in use as a
```

Comment:
Just out of curiosity, do any of these implementations track/limit by user IP address? Might be helpful in blocking/throttling any identified credential stuffers.

Reply:
One tracks by IP and can be used to reduce some of the credential stuffing capabilities but the problem with credential stuffing is that you don't need that many tries if you have specific credentials to try out.

Reply:
That's true, I was more thinking along the lines of rate limiting/blocking IPs that are trying out too many username+passwords. Might be an issue with public computers, but I'd think that a credential stuffer would go through username+passwords at a higher rate (with both successes and failures) than any shared device.